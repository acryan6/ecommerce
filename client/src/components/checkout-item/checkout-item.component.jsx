import React from "react";
import { connect } from "react-redux";

import {
  addItem,
  removeItem,
  clearItemFromCart,
} from "../../redux/cart/cart.actions";

import {
  CheckoutItemContainer,
  ImageContainer,
  Image,
  NameOrPrice,
  QuantityContainer,
  Arrow,
  Value,
  RemoveButton,
} from "./checkout-item.styles";

const CheckoutItem = ({ cartItem, addItem, removeItem, clearItem }) => {
  const { name, imageUrl, price, quantity } = cartItem;
  return (
    <CheckoutItemContainer>
      <ImageContainer>
        <Image src={imageUrl} alt="item" />
      </ImageContainer>
      <NameOrPrice>{name}</NameOrPrice>
      <QuantityContainer>
        <Arrow onClick={() => removeItem(cartItem)}>&#10094;</Arrow>
        <Value>{quantity}</Value>
        <Arrow onClick={() => addItem(cartItem)}>&#10095;</Arrow>
      </QuantityContainer>
      <NameOrPrice>{price}</NameOrPrice>
      <RemoveButton onClick={() => clearItem(cartItem)}>&#10005;</RemoveButton>
    </CheckoutItemContainer>
  );
};

const mapDispatchToProps = (dispatch) => ({
  clearItem: (item) => dispatch(clearItemFromCart(item)),
  addItem: (item) => dispatch(addItem(item)),
  removeItem: (item) => dispatch(removeItem(item)),
});

export default connect(null, mapDispatchToProps)(CheckoutItem);
