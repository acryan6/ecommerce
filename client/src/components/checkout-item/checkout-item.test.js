import React from "react";
import { shallow } from "enzyme";

import { CheckoutItem } from "./checkout-item.component";

describe("Checkout item", () => {
  let wrapper;
  let mockClearItem;
  let mockAddItem;
  let mockRemoveItem;

  beforeEach(() => {
    mockClearItem = jest.fn();
    mockAddItem = jest.fn();
    mockRemoveItem = jest.fn();

    const mockProps = {
      cartItem: {
        imageUrl: "mock.com",
        price: 0.01,
        name: "hats",
        quantity: 101,
      },
      clearItem: mockClearItem,
      addItem: mockAddItem,
      removeItem: mockRemoveItem,
    };

    wrapper = shallow(<CheckoutItem {...mockProps} />);
  });

  it("should render ChekoutItem component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call clearItem when remove button is clicked", () => {
    wrapper.find("RemoveButton").simulate("click");
    expect(mockClearItem).toHaveBeenCalled();
  });

  it("should call addItem when right arrow is clicked", () => {
    wrapper.find("QuantityContainer").childAt(2).simulate("click");
    expect(mockAddItem).toHaveBeenCalled();
  });

  it("should call removeItem when left arrow is clicked", () => {
    wrapper.find("QuantityContainer").childAt(0).simulate("click");
    expect(mockRemoveItem).toHaveBeenCalled();
  });
});
