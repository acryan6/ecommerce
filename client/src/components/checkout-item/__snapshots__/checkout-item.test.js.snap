// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout item should render ChekoutItem component 1`] = `ShallowWrapper {}`;
